tests/unit/clients/test_google_drive_client.py

```diff
@@ -145,7 +154,9 @@ def test_scandir_folder(self, mocker, folder_descriptor, url):
         client._service = mocker.MagicMock()
         with client:
             client.scandir()
-        lister.return_value.list.assert_called_once_with(url_base="gdrive://my drive/folder/")
+
+        kwargs = lister.return_value.mock_calls[0][2]
+        kwargs["url_base"] = "gdrive://my drive/folder/"
```

Comment:
i don't think this is testing anything - just setting the value in the dict at the end. since the url_base has moved to the constructor, did you mean something like
```suggestion
        assert lister.mock_calls[0][2]["url_base"] == "gdrive://my drive/folder/"
        assert lister.return_value.mock_calls[0][2] == {}
```

maybe the second assert is unnecessary though

Reply:
I dislike the mocking library for the lack of clarity in these things. As `lister` here is the class I'm actually testing that the constructor is called with `url_base`, not the `list` method.

Reply:
but is it actually testing anything given that it's missing the assert? ie if you change line 159 to 
```
kwargs["url_base"] = "some_random_value"
```
the test still passes

Reply:
oh crap! haha, :facepalm: :facepalm: :facepalm: XD